arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentCollectionViewController.swift

```diff
@@ -23,70 +23,84 @@ class CustomFlowLayout:UICollectionViewFlowLayout {
+        definesPresentationContext = false
     }
-
-    //TODO: implement this
-//    func isSuggestionsTableVisible() -> Bool {
-//        return (self.headerView?.suggestionsTableHeightConstraint?.constant == 0 ? false : true) ?? false
-//    }
-    
-    //MARK: - samples lookup by name
     
-    /// Returns the samples in all categories whose names are members of the
-    /// given names collection.
-    ///
-    /// - Parameter names: A collection of strings for which to test names
-    /// against.
-    /// - Returns: An array of samples whose names are members of `names1.
-    func samplesByNames<C: Collection>(_ names: C) -> [Sample] where C.Element == String {
-        return categories
-            .flatMap { $0.samples }
-            .filter { names.contains($0.name) }
+    private func addSearchController(){
+        
+        // create the view controller for displaying the search results
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let searchResultsController = storyboard.instantiateViewController(withIdentifier: "ContentTableViewController") as! ContentTableViewController
+        // search all samples in the app
+        searchResultsController.samples = categories.flatMap { $0.samples }
+        searchResultsController.containsSearchResults = true
+        
+        // create the search controller
+        let searchController = UISearchController(searchResultsController:searchResultsController)
+        searchController.obscuresBackgroundDuringPresentation = true
+        searchController.hidesNavigationBarDuringPresentation = false
+        // send search query updates to the results controller
+        searchController.searchResultsUpdater = searchResultsController
+        // retain a strong reference for iOS 10
+        self.searchController = searchController
+        
+        let searchBar = searchController.searchBar
+        searchBar.autocapitalizationType = .none
+        // set the color of "Cancel" text
+        searchBar.tintColor = .white
+        
+        if #available(iOS 11.0, *) {
+            // embed the search bar under the title in the navigation bar
+            navigationItem.searchController = searchController
+            
+            // find the text field to customize its appearance
+            if let textfield = searchBar.value(forKey: "searchField") as? UITextField {
+                // set the color of the insertion cursor
+                textfield.tintColor = UIColor.darkText
+                if let backgroundview = textfield.subviews.first {
+                    backgroundview.backgroundColor = UIColor.white
+                    backgroundview.layer.cornerRadius = 12
+                    backgroundview.clipsToBounds = true
+                }
+            }
+            
+        } else {
+            // embed the search bar in the title area of the navigation bar
+            navigationItem.titleView = searchController.searchBar
```

Comment:
It is unfortunate to see that this replaces the view's title. Is it possible to add the search bas as a section header? I've done something similar using a search bar a header view of a table view.

Reply:
I agree that this isn't ideal, but I didn't want to spend too much time writing version-specific code this since fewer and fewer users will be using versions of iOS prior to 11 going forward. I can revisit it if you really think it's worthwhile.

Reply:
I suppose it is fine the way it is.